cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/store/AppMetadataStore.java

```diff
@@ -1822,6 +1823,20 @@ private RunRecordDetail scanRunByReference(Collection<Field<?>> keys)
     }
   }
 
+  /**
+   * Deletes all completed run records with a start time before {@code timeUpperBound}, throwing
+   * {@code IOException} if the delete operation fails.
+   */
+  public void deleteRunsStartedBefore(Instant timeUpperBound) throws IOException {
```

Comment:
Since it's deleting `completed` records, wonder if it makes more sence to name the method `deleteCompletedRunsStartedBefore`?

Reply:
Also seems like we don't need to pass in `Instant timeUpperBound` because all we care about is the endTime and `AppMetadataStore` should be agnostic to the passed `Instant` from `RunRecordTimeToLiveService`?

How about make the method signature as `deleteCompletedRunsStartedBefore(long endTime)`

Reply:
A long would be more consistent with the rest of the class, but an Instant is a little cleaner, as we don't need to worry about if it's a timestamp in seconds or millis or anything like that. Took me a little while to look through the code to see if the timestamp is seconds or millis.

Reply:
I assumed this class just predated Java 8. I would lightly push for keeping it as an Instant so the precision is clear but if a long is preferred for consistency I can change this, let me know.

> Since it's deleting completed records, wonder if it makes more sence to name the method deleteCompletedRunsStartedBefore?

You're right, specifying completed runs in the function signature makes total sense, updated.